src/library/scala/concurrent/Done.scala

```diff
@@ -0,0 +1,36 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2003-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package scala.concurrent
+
+/**
+ * Typically used together with `Future` to signal completion when
+ * there is no actual value completed. More clearly signals intent
+ * than `Unit` and is available both from Scala and Java (which `Unit` is not).
+ * Also helps avoiding [[http://www.scala-lang.org/files/archive/spec/2.11/06-expressions.html#value-discarding value discarding]]
+ * pitfalls such as:
+ * {{{
+ *    // Scenario: inadvertent use of `map` instead of `flatMap` for `Future` composition
+ *    //
+ *    // compiles just fine: `map[T]` is infered to be `map[Unit]`, then `Future { println("two"); Done }` which
+ *    // is of type `Future[Done]` is discarded to satisfy `map[Unit]`
+ *    def compute: Future[Unit] = Future[Done] { println("one"); Done }.map( _ => Future { println("two"); Done } )
+ *
+ *    // does not compile!
+ *    def compute: Future[Done] = Future[Done] { println("one"); Done }.map( _ => Future { println("two"); Done } )
+ * }}}
+ */
+sealed abstract class Done
```

Comment:
@viktorklang No, I agree this isn't idiomatic scala code, and as @sjrd pointed out, nor is `getInstance`.

I suggested the inclusion of `Done` within the scala lib because it seemed useful beyond Akka's boundaries. That is, in libraries that may not have an Akka dependency. As it stands now, those libs will become annoying to integrate with Akka code that is making use of `Done`. Think `futureOfUnit.map(_ => Done)`.

Now this is all about setting the limit for these hacks so that it gets accepted by both Akka and Scala. IMO we're fine with the `sealed abstract class` but not with the `require(reflectionHack)`.
Maybe someone who's more deeply involved with the scala lib could say something about this?